### DISCUSSION TITLE
Produces a duplicate `end` in Lua

### DISCUSSION THREAD

**Discussion by sumneko** [community/community#12415] [Copilot Conversations]
![copilot-improve](https://user-images.githubusercontent.com/5213431/156810568-f6a65cb1-b0a9-4291-ac2d-ea69d8e4e56d.gif)

When it works inside a complete `if then end` block, it sometimes tries to generate an `end` again.

**Comment by Adrian-Samoticha** [comment]
I've had similar issues in other languages as well, so this issue is definitely not limited to Lua.

**Comment by TinusNL** [comment]
I am experiencing the same issue. It just wants to add an extra end even if it is already placed. 
![image](https://user-images.githubusercontent.com/66055492/161000868-2f2ca2c9-455e-4237-910c-7833445a1976.png)

**Reply by Urethramancer** [reply]
I had the exact same behaviour in Go this weekend. Seems it's a new habit it's picked up lately. Still getting superfluous ") after some printfs too.

**Reply by ahey** [reply]
This is happening with elixir too, a superfluous `end` is suggested

**Comment by CharlotteAltima** [comment]
I have this problem with block comments in C and C++.
![duplicate_end_comment](https://user-images.githubusercontent.com/88128354/161533028-cbb1620c-9d13-41fa-a5f4-b1dbab8d85af.png)

**Comment by MillhioreBT** [comment]
Confirmed I also have this problem with several languages, it's really cumbersome.
